### ISSUE TITLE
generate duplicate code

### ISSUE THREAD

**Issue by Jiang-Le** [switchupcb/copygen#18]
Please provide the following information.

### YML

```yml
generated:
  setup: ./setup.go
  output: ../copygen.go

```

### Go

```go
// setup.go

package setup

import (
	"github.com/switchupcb/copygen/examples/oboe/domain"
	"github.com/switchupcb/copygen/examples/oboe/model"
)

type Copygen interface {
	ModelsToDomain(model.Adgroup, model.Campaign) domain.Adgroup
}

// domain.go 

package domain

type Adgroup struct {
	ID   int64
	Name string
	Planning Planning
	Promotion Promotion
	ResourcePos ResourcePos
}

type Promotion struct {
	PromotionType int32
}

type Planning struct {
	PlanningID string
}

type ResourcePos struct {
	PlacementIDs []string
}

// model.go

package model

type Adgroup struct {
	ID           int64
	Name         string
	PlanningID   string
	PlacementIDs []string
}

type Campaign struct {
	PromotionType int32
}

```

### Generation

```go
// Code generated by github.com/switchupcb/copygen
// DO NOT EDIT.

package setup

import (
	"github.com/switchupcb/copygen/examples/oboe/domain"
	"github.com/switchupcb/copygen/examples/oboe/model"
)

// ModelsToDomain copies a Adgroup, Campaign to a Adgroup.
func ModelsToDomain(tA domain.Adgroup, fA model.Adgroup, fC model.Campaign) {
	// Adgroup fields
	tA.ResourcePos.PlacementIDs = fA.PlacementIDs
	tA.ResourcePos.PlacementIDs = fA.PlacementIDs
	tA.Promotion.PromotionType = fC.PromotionType
	tA.ResourcePos.PlacementIDs = fA.PlacementIDs
	tA.ResourcePos.PlacementIDs = fA.PlacementIDs
	tA.Promotion.PromotionType = fC.PromotionType
	tA.Planning.PlanningID = fA.PlanningID
	tA.Name = fA.Name
	tA.ID = fA.ID

}

```

## Environment
**windows**: windows
**Copygen Version**: v0.2.4

**Comment by Jiang-Le** [comment]
by the way, if `map` option is used, the Automatcher will does work. 
This feature doesn't seem reasonable. why not use Automatcher default, and `map` option can override some field copy.

**Comment by switchupcb** [comment]
@Jiang-Le This issue is known. I've started fixing it right now!

The reason the `map` option prevents the automatcher is because it is assumed a user using the match option may not want to have auto-matched fields. This will be fixed within a day or two with the addition of the `automatch` option.

**Comment by Jiang-Le** [comment]
thank you for your reply.
the more import issue is duplicated code is generated in my case.

the generated code is
```go
// Code generated by github.com/switchupcb/copygen
// DO NOT EDIT.

package setup

import (
	"github.com/switchupcb/copygen/examples/oboe/domain"
	"github.com/switchupcb/copygen/examples/oboe/model"
)

// ModelsToDomain copies a Adgroup, Campaign to a Adgroup.
func ModelsToDomain(tA domain.Adgroup, fA model.Adgroup, fC model.Campaign) {
	// Adgroup fields
	tA.ResourcePos.PlacementIDs = fA.PlacementIDs
	tA.ResourcePos.PlacementIDs = fA.PlacementIDs
	tA.Promotion.PromotionType = fC.PromotionType
	tA.ResourcePos.PlacementIDs = fA.PlacementIDs
	tA.ResourcePos.PlacementIDs = fA.PlacementIDs
	tA.Promotion.PromotionType = fC.PromotionType
	tA.Planning.PlanningID = fA.PlanningID
	tA.Name = fA.Name
	tA.ID = fA.ID

}
```

**Comment by switchupcb** [comment]
@Jiang-Le 
Fixed in #12:
```
go install github.com/switchupcb/copygen@main
```
```go
package setup

import (
        "github.com/switchupcb/copygen/examples/issue/domain"
        "github.com/switchupcb/copygen/examples/issue/model"
)

// ModelsToDomain copies a Adgroup, Campaign to a Adgroup.
func ModelsToDomain(tA domain.Adgroup, fA model.Adgroup, fC model.Campaign) {
        // Adgroup fields
        tA.ID = fA.ID
        tA.Name = fA.Name
        tA.Planning.PlanningID = fA.PlanningID
        tA.Promotion.PromotionType = fC.PromotionType
        tA.ResourcePos.PlacementIDs = fA.PlacementIDs
}
```

_You can use pointers on `model.Adgroup` and `model.Campaign` to avoid allocations._

**Comment by switchupcb** [comment]
You may experience a regression being fixed in v0.3. That will come out sometime today.
